packages/components/src/internal/app/utils.ts

```diff
@@ -222,57 +224,223 @@ export function getMenuSectionConfigs(user: User, currentApp: string): List<Map<
+
+    if (isSMPrimary || currentAppProperties?.productId === SAMPLE_MANAGER_APP_PROPERTIES.productId) {
+        if (storageConfig) {
+            sectionConfigs = sectionConfigs.push(Map({[FREEZERS_KEY]: storageConfig}));
+        }
         sectionConfigs = sectionConfigs.push(
             Map({
-                [WORKFLOW_KEY]: new MenuSectionConfig({
-                    headerURL: smAppBase + WORKFLOW_HOME_HREF.toHref(),
-                    iconURL: imageURL('_images', 'workflow.svg'),
-                    seeAllURL: smAppBase + AppURL.create(WORKFLOW_KEY).toHref(),
-                }),
-                [USER_KEY]: new MenuSectionConfig({
-                    iconCls: 'fas fa-user-circle ',
-                }),
+                [WORKFLOW_KEY]: workflowConfig,
+                [USER_KEY]: USER_SECTION_CONFIG,
             })
         );
+    } else if (isBioPrimary) {
+        if (isRequestsEnabled(moduleContext)) {
+            // When "Requests" are enabled render as two columns
+            let requestsCol = Map({
+                [REQUESTS_KEY]: REQUESTS_SECTION_CONFIG,
+            });
+            // ... and put the storage in this same column
+            if (storageConfig) {
+                requestsCol = requestsCol.set(FREEZERS_KEY, storageConfig);
+            }
+            sectionConfigs = sectionConfigs.push(
+                requestsCol,
+                Map({
+                    [WORKFLOW_KEY]: workflowConfig,
+                    [MEDIA_KEY]: getMediaSectionConfig(appBase),
+                    [NOTEBOOKS_KEY]: getNotebooksSectionConfig(appBase),
+                })
+            );
+        } else {
+            if (storageConfig) {
+                sectionConfigs = sectionConfigs.push(
+                    Map({[FREEZERS_KEY]: storageConfig})
+                );
+            }
+            sectionConfigs = sectionConfigs.push(
+                Map({
+                    [WORKFLOW_KEY]: workflowConfig,
+                    [MEDIA_KEY]: getMediaSectionConfig(appBase),
+                    [NOTEBOOKS_KEY]: getNotebooksSectionConfig(appBase),
+                })
+            );
+        }
     } else {
-        const userSectionConfig = new MenuSectionConfig({
-            iconCls: 'fas fa-user-circle ',
-        });
-        sectionConfigs = sectionConfigs.push(Map<string, MenuSectionConfig>().set(USER_KEY, userSectionConfig));
+        if (storageConfig) {
+            sectionConfigs = sectionConfigs.push(Map({[FREEZERS_KEY]: storageConfig}));
+        }
+        sectionConfigs = sectionConfigs.push(Map ({[USER_KEY]: USER_SECTION_CONFIG}));
     }
     return sectionConfigs;
 }
 
-function getProductId(moduleName: string): string {
+export const useMenuSectionConfigs = (user: User, appProperties: AppProperties, moduleContext: any): List<Map<string, MenuSectionConfig>> => {
+    return useMemo(() => {
+        return getMenuSectionConfigs(user, appProperties.productId, moduleContext);
+    }, [user, moduleContext]);
```

Comment:
should `appProperties.productId` also be in this dependency array?

Reply:
Good catch.  Thanks.